src/EGL/icd_enumeration.md

```diff
@@ -0,0 +1,113 @@
+# EGL ICD enumeration
+
+EGL can't rely on asking an X server for a vendor name like GLX can, so
+instead, it enumerates and loads every available vendor library. Loading every
+vendor is also needed to support extensions such as
+EGL\_EXT\_device\_enumeration.
+
+In order to find the available vendor libraries, each vendor provides a JSON
+file in a well-known directory, similar to how Vulkan ICDs are loaded.
+
+## ICD discovery
+
+* All environment variables mentioned here are ignored (treated as though
+    they were unset) if the process is setuid.
+
+* If the environment variable `__EGL_VENDOR_LIBRARY_FILENAMES` is set,
+    it is a colon-separated list of JSON filenames. The ICDs described in
+    those files are loaded, in the order given. No other ICDs are loaded.
+
+* Otherwise, if the environment variable `__EGL_VENDOR_LIBRARY_DIRS`
+    is set, it is a colon-separated list of directories. Each directory in
+    turn is scanned for files named `*.json` and the ICDs that they describe
+    are loaded. Files in the same directory are loaded in `strcmp()` order,
+    so `40_myvendor.json` is considered to be higher-priority than
+    `50_yourvendor.json`.
+
+* If neither environment variable is set, GLVND behaves
+    as though `__EGL_VENDOR_LIBRARY_DIRS` had been set to
+    `${sysconfdir}/glvnd/egl_vendor.d:${datadir}/glvnd/egl_vendor.d`,
+    replacing `${sysconfdir}` and `${datadir}` with the values that were
+    set when GLVND was compiled.
+
+* Each JSON file describing an ICD must have a JSON object at top level.
+    * The key `file_format_version` must have a string value giving the
+        file format `major.minor.micro` version number. This specification
+        describes version `1.0.0`. Versions 1.0.x are required to be
+        compatible with this specification, in the sense that an EGL loader
+        that only implements file format version 1.0.0 will load all version
+        1.0.x JSON files successfully. Different major and minor versions
+        might require loader changes.
+    * The key `ICD` must have an object value.
+        * In the `ICD` object, the key `library_path` must have a string value.
+            * If the library path is a bare filename with no directory
```

Comment:
For a bare filename and an absolute path, I think it's better to just say that the path is passed directly to dlopen, since the dlopen documentation will be more complete than anything we're likely to write up here.

Something like this:
If the library path is either an absolute pathname or a bare filename with no directory separators, then the path is passed directly to dlopen. See the documentation for dlopen for the full details on how it searches for a library.

Reply:
I was deliberately leaving this vague, so that we can change it to be consistent with Vulkan without contradicting the specification that I'd just written, after you said on #187 that you'd consider that change (I haven't implemented it yet, sorry).

Reply:
Oh, sorry, I misread - you were talking about the other two cases. OK.